### ISSUE TITLE
Can a trace description be blank?

### ISSUE THREAD

**Issue by AntonKhorev** [openstreetmap/openstreetmap-website#3598]
### URL

https://www.openstreetmap.org/traces/new

### How to reproduce the issue?

1. go to *new trace* form
2. choose a file, leave everything else blank
3. try to submit, receive the message: *Description can't be blank, is too short (minimum is 1 character)*
4. choose a file again, enter a single space character in the description
5. try to submit, receive the message: *Description can't be blank*
6. choose a file again, enter a valid description
7. submit successfully
8. open the newly submitted trace and edit it
9. replace the description with a single space character
10. submit successfully

What was the point of rejecting a blank description on step 3 if it can be made blank by editing?

**Comment by AntonKhorev** [comment]
commit allowing blank descriptions: https://github.com/openstreetmap/openstreetmap-website/commit/81021a108151270e1aef4570ded6f7aeed901722

**Comment by tomhughes** [comment]
Which explains exactly why in the commit comment - there are historic records with no description so we only require a description for newly created traces.

Now we could extend that validation to the edit action, which would prevent people removing the description, but that would also force the addition of a description if you were editing an old trace that didn't have one.

**Comment by gravitystorm** [comment]
I think we can probably change the validation to only trigger when the description changes during the update - i.e. allow the description to remain blank, but not to allow a description to be changed from something else to become blank.

**Comment by AntonKhorev** [comment]
Why do you need description to be non-blank if everything works when it's blank?

**Comment by AntonKhorev** [comment]
> So I think we should focus this issue on fixing the edge case that you found, rather than changing the expected behaviour. If there's a widespread desire to change descriptions to be completely optional again then that can be discussed separately.

And the fix is to undo that commit: change `:presence => { :on => :create }` to `:presence => true` for [description validation](https://github.com/openstreetmap/openstreetmap-website/blob/48fff2bfeabb9a9bafd78c9cd4a0077e6569b631/app/models/trace.rb#L48)?

**Comment by tomhughes** [comment]
No, because that would reintroduce the original bug that you wouldn't be able to make any changes to an old trace that had a blank description.

What @gravitystorm means is to find a way to stop an existing description being removed without causing problems for traces that have never had a description.

**Comment by AntonKhorev** [comment]
What's the problem with adding a description along with these changes? What kind of changes you expect? Do you think that these old traces are going to be changed? The length requirement (at least one character) is present even for the update, won't the empty descriptions fail it anyway? If update is done through the API doesn't it require the description to be present?

The problematic scenario I can think of: someone deciding to manually change visibility of a bunch of old traces who is going to find that they also need to type something in the description field. I don't think it's likely to happen.

**Comment by AntonKhorev** [comment]
I tried setting trace description to an empty string and editing the trace without changing the description. When there's no tags it gets rejected as too short. When there are tags it crashes when trying to update them. So you can't keep empty descriptions when editing. However you can keep nonempty blank descriptions like a single space, but do you want to? Do they exist in any significant quantity in old traces?
